providers/common/include/prov/fipsindicator.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023-2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifdef FIPS_MODULE
+
+# define OSSL_FIPS_INDICATOR_ID_STRICT_CHECK0 0
+# define OSSL_FIPS_INDICATOR_ID_STRICT_CHECK1 1
+# define OSSL_FIPS_INDICATOR_ID_STRICT_CHECK2 2
+
+# include <openssl/core.h> /* OSSL_CALLBACK, OSSL_LIB_CTX */
+# include <openssl/indicator.h>
+
+typedef struct ossl_fips_indicator_st {
+    int approved;               /* -1 initially, can be set to 1 or 0 */
+    unsigned char strict_checks_set;  /* A bit mask - each bit that is 0 */
+    unsigned char strict_checks;      /* a bitmask - each bit is 1 or 0 */
+} ossl_FIPS_INDICATOR;
+
+typedef int (OSSL_FIPS_CHECK_CB)(OSSL_LIB_CTX *libctx);
+
+int ossl_FIPS_INDICATOR_callback(OSSL_LIB_CTX *libctx, const char *type,
+                                 const char *desc);
+
+void ossl_FIPS_INDICATOR_init(ossl_FIPS_INDICATOR *ind);
+void ossl_FIPS_INDICATOR_set_approved(ossl_FIPS_INDICATOR *ind, int approved);
+int ossl_FIPS_INDICATOR_get_approved(const ossl_FIPS_INDICATOR *ind);
+void ossl_FIPS_INDICATOR_set_strict(ossl_FIPS_INDICATOR *ind, int id, int enable);
+int ossl_FIPS_INDICATOR_get_strict(const ossl_FIPS_INDICATOR *ind, int id);
+
+/* Place this in the algorithm ctx structure */
+# define OSSL_FIPS_INDICATOR_DECLARE() ossl_FIPS_INDICATOR indicator;
```

Comment:
Dropping the ';' is a bit harder since in the #else where it is defined as nothing it will not sit too well in the struct if the code is { OSSL_FIPS_INDICATOR_DECLARE; }